crux_core/src/bridge/mod.rs

```diff
@@ -0,0 +1,83 @@
+mod registry;
+pub mod step_serde;
+
+use serde::{Deserialize, Serialize};
+
+use crate::{App, Core, Effect};
+use registry::ResolveRegistry;
+
+/// Request for a side-effect passed from the Core to the Shell. The `uuid` links
+/// the `Request` with the corresponding call to [`Core::response`] to pass the data back
+/// to the [`App::update`] function (wrapped in the event provided to the capability originating the effect).
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Request<Eff>
+where
+    Eff: Effect,
+{
+    pub uuid: Vec<u8>,
+    pub effect: Eff::Ffi,
+}
+
+pub struct Bridge<'de, Eff, A>
```

Comment:
I suspect you're going to have problems with this lifetime when you come to try and use the `Bridge`, but probably solvable.

Reply:
Yes, I suspect so, I'm getting to that now, so I'll see what happens.

Reply:
I spent a little bit of time trying to use the `Bridge` in `hello_world` and this is true—this lifetime conflicts with the `static` lifetime we use on `Core` (or now `Bridge`) in the shell.

Reply:
@wasnotrice Try again with the latest. I have the Counter example working on iOS now. Will review the new Rust public API for sensible naming and ergonomics, and then migrate all the Rust examples across.

Reply:
@charypar nice! I was able to get `hello_world` to compile, though the tests still need some work